oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/invalidData/InvalidIndexDefinitionTest.java

```diff
@@ -0,0 +1,259 @@
+        def.setProperty(LuceneIndexConstants.COMPAT_MODE, 3);
+        root.commit();
+    }
+    
+    @Test(expected = PatternSyntaxException.class)
+    public void invalidValueRegex() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // An incorrect value, for example "[a-z", results in
+        // java.util.regex.PatternSyntaxException: Unclosed character class near index 3
+        def.setProperty(LuceneIndexConstants.PROP_VALUE_REGEX, "[a-z");
+        root.commit();
+    }
+    
+    @Test(expected = PatternSyntaxException.class)
+    public void invalidQueryFilterRegex() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // An incorrect value, for example "[a-z", results in
+        // java.util.regex.PatternSyntaxException: Unclosed character class near index 3
+        def.setProperty(LuceneIndexConstants.PROP_QUERY_FILTER_REGEX, "[a-z");
+        root.commit();
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidBlobSize() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // 1L + Integer.MAX_VALUE results in IllegalArgumentException: Out of range: 2147483648
+        def.setProperty("blobSize", 1L + Integer.MAX_VALUE);
+        root.commit();
+    }    
+
+    @Test
+    public void negativeBlobSize() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // 1L + Integer.MAX_VALUE results in IllegalArgumentException: Out of range: 2147483648
+        def.setProperty("blobSize", -1);
+        root.commit();
+        String query = "select [jcr:path] from [nt:base] where isdescendantnode('/tmp') and upper([test]) = 'HELLO'";
+        assertThat(explain(query), containsString("lucene:test"));
+        assertQuery(query, Lists.newArrayList("/tmp/testNode"));
+    }    
+
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidMaxFieldLength() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // 1L + Integer.MAX_VALUE results in IllegalArgumentException: Out of range: 2147483648
+        def.setProperty(FulltextIndexConstants.MAX_FIELD_LENGTH, 1L + Integer.MAX_VALUE);
+        root.commit();
+    }    
+
+    @Test
+    public void invalidEvaluatePathRestriction() throws CommitFailedException {
+        Tree def = createIndexNodeAndData();
+        // errors here are ignored
+        def.setProperty(LuceneIndexConstants.EVALUATE_PATH_RESTRICTION, "abc");
+        root.commit();
+        String query = "select [jcr:path] from [nt:base] where isdescendantnode('/tmp') and upper([test]) = 'HELLO'";
+        assertThat(explain(query), containsString("lucene:test"));
+        assertQuery(query, Lists.newArrayList("/tmp/testNode"));
+    }
+
+    @Test
+    public void invalidIncludedPath() throws CommitFailedException {
+        // this will commit, but in oak-run it will not work:
```

Comment:
Maybe we can add assertions on logs here to check this error is logged properly ? 
Refer LogCustomizer usage here https://github.com/apache/jackrabbit-oak/blob/trunk/oak-core/src/test/java/org/apache/jackrabbit/oak/query/stats/QuerySimilarCostTest.java